src/app/vault/ciphers.component.ts

```diff
@@ -81,6 +89,79 @@ export class CiphersComponent extends BaseCiphersComponent implements OnDestroy
     this.didScroll = this.pagedCiphers.length > this.pageSize;
   }
 
+  async loadOrganizationNames(): Promise<Map<string, string>> {
+    const organizations = await this.organizationService.getAll();
+    const organizationNames = new Map<string, string>();
+
+    organizations.forEach((organization) => {
+      organizationNames.set(organization.id, organization.name);
+    });
+
+    return organizationNames;
+  }
+
+  getOrganizationById(c: CipherView): string {
+    return c.organizationId != null
+      ? this.organizationNames.get(c.organizationId)
+      : this.i18nService.t("myVault");
+  }
+
+  sortCiphers(sortBy: string): void {
+    if (sortBy === this.sortBy) {
+      this.sortedDescending = !this.sortedDescending;
+    } else {
+      this.sortBy = sortBy;
+      this.sortedDescending = true;
+    }
+
+    if (sortBy === "lastEdited") {
+      if (this.sortedDescending) {
+        this.ciphers.sort((a: CipherView, b: CipherView) => {
+          return b.revisionDate.valueOf() - a.revisionDate.valueOf();
+        });
+      } else {
+        this.ciphers.sort((a: CipherView, b: CipherView) => {
+          return a.revisionDate.valueOf() - b.revisionDate.valueOf();
+        });
+      }
+      this.showData = "lastEdited";
```

Comment:
I don't think you need this line (and the equivalent in each `else if` block) because we set `showData` separately now. You're only ever sorting the data that you've already displayed.